### DISCUSSION TITLE
REDB backend

### DISCUSSION THREAD

**Discussion by casey** [bitcoindevkit/bdk#691] [Ideas]
[redb](https://github.com/cberner/redb/) is a pure-Rust key-value store, and would be a good candidate for an alternative back-end. It is secure against data loss in some cases which sled probably isn't. It is still alpha quality, and the on-disk format has not been finalized, so it might not be quite ready yet, but could be good to watch for the future.

I mention this because I had issues building a binary on a Linux box due to the SQLite backend, and I don't personally believe that sled is mature enough to be used for production workloads, so another option would be nice.

**Comment by evanlinjin** [comment]
This doesn't seem too difficult to implement for first time contributors imo ☺️🤘

**Comment by 110CodingP** [comment]
Hi @casey just published a [crate](https://crates.io/crates/bdk_redb)  for the same, would be great if you could help review!
